core/src/test/scala/org/apache/spark/storage/StorageStatusListenerSuite.scala

```diff
@@ -150,4 +158,12 @@ class StorageStatusListenerSuite extends FunSuite {
     listener.onUnpersistRDD(SparkListenerUnpersistRDD(1))
     assert(listener.executorIdToStorageStatus("big").numBlocks === 0)
   }
+  
+  test("Killed Executor Entry removed after configurable time") {
+    val localtestconf = new SparkConf().set(StorageStatusListener.TIME_TO_EXPIRE_KILLED_EXECUTOR,"5s")
+    val listener = new StorageStatusListener(localtestconf)
+    listener.removedExecutorIdToStorageStatus.put("1", new StorageStatus(null, 50))
+    Thread.sleep(5500)
```

Comment:
I think you can do something in between -- StorageStatusListener can have a `private[storage]` constructor which takes the ticker, and the public one just defaults it to the system ticker.  Yes, you would not be testing *exactly* the same behavior, but it tests the important parts.

Sleeping isn't the worst thing in this case -- often it leads to flaky tests, though I don't think that would be the case here.  Still, 5 seconds is awfully long for this test when it should take a tiny fraction of that, and it adds up over all the tests.